Schemas.md

```diff
@@ -0,0 +1,521 @@
+
+* `validate(handler_args, handler_args_schemas)` in `payload_validator.py`: This method is the core method for SVS functionality. It collects all the AssertionErrors raised from schema_utils.
+  * `handler_args`: The arguments from payload/ request.
+  * `handler_args_schemas`: Schema from the handler class.(See [this link](#how-to-write-validation-schema-for-handlers) for more information on how to write a schema).
+
+* `normalize_against_schema(obj, schema)` in `schema_utils.py`: This method normalizes the obj against its schema and raises `AssertionError` if any of the validation checks fail. This `AssertionError` is represented as `InvalidInputException` to users.
+  * `obj`: The object which needs to be normalized.
+  * `schema`: The schema for the object.
+
+#### How to write validation schema for handlers
+
+If you’re writing a new handler method, you’ll need to add schema validation for the handler args. To do this, follow the steps below:
+
+1. **List all the arguments passed to each method in the handler**
+   Make a list of all the arguments passed to each method in the handler class. Arguments received by a handler class method can be categorized into 3 types:
+
+   * **URL path elements** : The data which is present as a part of the URL are called URL path elements. Example: in ```url/<exploration_id>/```, the exploration_id is a URL path element.
+   * **Payload arguments**: The data which comes from payloads are called payload arguments. These data are typically received by PUT and POST methods.
+   * **URL query parameters**: Query parameters are a defined set of parameters attached to the end of a url. They are extensions of the URL that are used to help define specific content or actions based on the data being passed. Example: in ```url/<exploration_id>?username=nikhil```, there is a single URL query parameter, with arg name “username” and value “nikhil”. URL query parameters are typically received by GET and DELETE methods.
+
+   If you face any difficulty see the [debugging section](#debugging-tricks) or reach out to any of the persons mentioned in the [contact section](#contact).
+
+2. **Determine the schema for each argument**
+   For writing schema each argument should be analysed deeply, like the use of argument in the backend structure of the code and based on the analysis, schema for the arguments should be written by following the [boilerplate code](#handlers-with-no-arguments). See these links for more information on [allowed schema keys](#write-a-schema), [Important code pointers](#important-code-pointers), and [examples](#examples-for-reference).
```

Comment:
I think the grammar/structure of this could be improved.

Reply:
done